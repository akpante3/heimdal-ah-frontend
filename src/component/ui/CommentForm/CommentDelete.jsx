import React from 'react';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';
import { deleteArticleComment } from '../../../actions/CommentActions/CommentActions';
import { deleteCommentReplies } from '../../../actions/ReplyActions/ReplyAction';

/**
  * renderComponent
  * @method Function Function based Component
  * @summary React component for rendering the card components
  * @param {Object} props - React PropTypes
  * @property {String|Object} className - String className compatible object for styling
  * @return {Node} React node containing delete comment and replies
  */
const DeleteButton = (props) => {
  const deleteComment = (e) => {
    e.preventDefault();
    const { commentId, buttonName, replyId } = props;
    if (buttonName === 'comment') {
      props.deleteArticleComment(1, commentId);
    } else {
      props.deleteCommentReplies(commentId, replyId);
    }
  };

  /**
     * Renders the component.
     *
     * @memberof app.components.Comment
     * @return {string} - HTML markup for the component
   */

  return (
    <button className='archive_button btn btn-danger' onClick={e => deleteComment(e)} type='submit'><FontAwesome name='trash' className='' /></button>
  );
};

/**
 * @method module:Reactator.ReduxContainerBuilderMapStateToProps
 *
 * @param {function} mapStateToProps - function mapping redux state to props
 *
 * @return {ReduxContainerBuilder} this builder
*/
const mapStateToProps = state => ({
  comment: state.comment,
});

export default connect(
  mapStateToProps,
  {
    deleteArticleComment, deleteCommentReplies,
  },
)(DeleteButton);
