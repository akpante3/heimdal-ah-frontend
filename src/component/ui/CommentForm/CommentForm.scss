@import '../../../styles/variables.scss';

.form__comment {
  font-family:$font_family_reem-kufi;
  border-top: 1px solid $heimdal-comment-border;
  
  .comment__head__a {
    font-size: 25px;
    margin-top: 60px;
    padding-top: 70px;
  }
  .comment__head__b {
    font-size: 20px;
  }
  .comment-form {
    border: 1px solid $heimdal-comment-border;
    border-radius: 0!important; 
    margin-right: 70px!important;
    width: 100%;
  }
  
  .flex-container__a {
    display: flex;
    justify-content: flex-end;
    margin-top: 10px!important;
  }
  .flex-container__b {
    display: flex;
    justify-content: space-between;
  }
  
  .comment_check {
    margin-right: 30px!important;
    margin-top: 8px;

  }

  .private__comment {
    margin-right: 50px;
    margin-top: 10px;
    font-size: 15px;
    color: $heimdal-private-comment-color;
    font-family: $font-family__cardo;
  }
}

