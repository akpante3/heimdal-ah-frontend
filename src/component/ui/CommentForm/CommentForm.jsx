/* eslint-disable react/no-unused-prop-types */
/* eslint-disable react/forbid-prop-types */
import React, { Component } from 'react';
import { FormGroup, FormControl } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Button from '../Buttons/Button';
import { postArticleComment } from '../../../actions/CommentActions/CommentActions';
import Alert from '../Alert/Alert';
import './CommentForm.scss';

/**
  * renderComponent
  * @method Class Class based Component
  * @summary React component for rendering the card components
  * @param {object}  comment contains the comments of an article
  * @param {Object} props - React PropTypes
  * @property {String|Object} className - String className compatible object for styling
  * @return {Node} React node containing comment form view
  */
export class CommentForm extends Component {
  state = {
    formIsValid: false,
    commentError: '',
    content: '',
    checkbox: false,
  }

   onChange = (e) => {
     this.setState({
       [e.target.name]: e.target.value,
       commentError: '',
     });
   }

   onBlur = (e) => {
     if (e.target.value === '') {
       this.setState({
         commentError: 'Content cannot be Empty',
       });
     } else {
       this.setState({
         formIsValid: true,
         commentError: '',
       });
     }
   }

   onCheck = () => {
     const { checkbox } = this.state;
     this.setState({
       checkbox: !checkbox,
       commentError: '',
     });
   }

   submitForm = (e) => {
     e.preventDefault();
     const { user } = this.props.auth;
     const { formIsValid, content, checkbox } = this.state;
     if (content.length >= 1500) {
       this.setState({
         formIsValid: false,
         commentError: "can't have more than 1500 characters",
       });
     } else if (formIsValid) {
       const data = {
         content,
         isPrivate: checkbox,
       };

       this.props.postArticleComment(this.props.articleId, data, user.image);
       this.setState({
         content: '',
       });
     } else {
       this.setState({
         commentError: 'Content cannot be Empty',
       });
     }
   }

   /**
     * Renders the component.
     *
     * @memberof app.components.Comment
     * @return {string} - HTML markup for the component
   */
   render() {
     const { checkbox, commentError } = this.state;
     return (
       <div className='container form__comment'>
         <div className='comment__head'>
           <p className='comment__head__a'>ADD COMMENT</p>
           <p className='comment__head__b'>Comment</p>
         </div>
         <div>
           {' '}
           {commentError !== '' && <Alert type='warning' message={this.state.commentError} title='Invalid Credenntials' /> }
           <div />
         </div>
         <form className='' onSubmit={this.submitForm}>
           <FormGroup controlId="formControlsTextarea">
             <FormControl
               onBlur={this.onBlur}
               onChange={this.onChange}
               componentClass="textarea"
               value={this.state.content}
               name='content'
               className='comment-form'
               rows="7"
               cols="70"
               placeholder="textarea"
             />
           </FormGroup>
           <div className='flex-container__a'>
             <div className="form-check flex-container__b">
               <div className='private__comment'>Make it Private</div>
               <div className='comment_check'>
                 <input type="checkbox" name='checkbox' onChange={e => this.onCheck(e)} className="form-check" value={!checkbox} />
               </div>
             </div>
             <Button type="comment" className='comment-button' label="Add Comment" />
           </div>
         </form>
       </div>
     );
   }
}

CommentForm.propTypes = {
  postArticleComment: PropTypes.func.isRequired,
  comment: PropTypes.object.isRequired,
  auth: PropTypes.object.isRequired,
  error: PropTypes.object.isRequired,
};

/**
 * @method module:Reactator.ReduxContainerBuilderMapStateToProps
 *
 * @param {function} mapStateToProps - function mapping redux state to props
 *
 * @return {ReduxContainerBuilder} this builder
*/
const mapStateToProps = state => ({
  comment: state.comment,
  auth: state.auth,
  error: state.errors,
});

export default connect(mapStateToProps, { postArticleComment })(CommentForm);
