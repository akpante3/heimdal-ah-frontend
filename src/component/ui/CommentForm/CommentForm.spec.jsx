/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { CommentForm } from './CommentForm';

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      postArticleComment: jest.fn(),
      comment: '',
      auth: '',
      error: '',
    };
  });

  it('renders One Comment Form components', () => {
    wrapper = shallow(<CommentForm {...props} />);
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<CommentForm {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<CommentForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
