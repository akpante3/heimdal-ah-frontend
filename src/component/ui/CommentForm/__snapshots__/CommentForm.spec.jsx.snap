// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentForm
    auth=""
    comment=""
    error=""
    postArticleComment={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="comment__head"
        >
          <p
            className="comment__head__a"
          >
            ADD COMMENT
          </p>
          <p
            className="comment__head__b"
          >
            Comment
          </p>
        </div>,
        <div>
           
          <div />
        </div>,
        <form
          className=""
          onSubmit={[Function]}
        >
          <FormGroup
            bsClass="form-group"
            controlId="formControlsTextarea"
          >
            <FormControl
              bsClass="form-control"
              className="comment-form"
              cols="70"
              componentClass="textarea"
              name="content"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="textarea"
              rows="7"
              value=""
            />
          </FormGroup>
          <div
            className="flex-container__a"
          >
            <div
              className="form-check flex-container__b"
            >
              <div
                className="private__comment"
              >
                Make it Private
              </div>
              <div
                className="comment_check"
              >
                <input
                  className="form-check"
                  name="checkbox"
                  onChange={[Function]}
                  type="checkbox"
                  value={true}
                />
              </div>
            </div>
            <Button
              Class=""
              className="comment-button"
              label="Add Comment"
              onClick={[Function]}
              type="comment"
            />
          </div>
        </form>,
      ],
      "className": "container form__comment",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="comment__head__a"
            >
              ADD COMMENT
            </p>,
            <p
              className="comment__head__b"
            >
              Comment
            </p>,
          ],
          "className": "comment__head",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ADD COMMENT",
              "className": "comment__head__a",
            },
            "ref": null,
            "rendered": "ADD COMMENT",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Comment",
              "className": "comment__head__b",
            },
            "ref": null,
            "rendered": "Comment",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            false,
            <div />,
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FormGroup
              bsClass="form-group"
              controlId="formControlsTextarea"
            >
              <FormControl
                bsClass="form-control"
                className="comment-form"
                cols="70"
                componentClass="textarea"
                name="content"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="textarea"
                rows="7"
                value=""
              />
            </FormGroup>,
            <div
              className="flex-container__a"
            >
              <div
                className="form-check flex-container__b"
              >
                <div
                  className="private__comment"
                >
                  Make it Private
                </div>
                <div
                  className="comment_check"
                >
                  <input
                    className="form-check"
                    name="checkbox"
                    onChange={[Function]}
                    type="checkbox"
                    value={true}
                  />
                </div>
              </div>
              <Button
                Class=""
                className="comment-button"
                label="Add Comment"
                onClick={[Function]}
                type="comment"
              />
            </div>,
          ],
          "className": "",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": <FormControl
                bsClass="form-control"
                className="comment-form"
                cols="70"
                componentClass="textarea"
                name="content"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="textarea"
                rows="7"
                value=""
              />,
              "controlId": "formControlsTextarea",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "className": "comment-form",
                "cols": "70",
                "componentClass": "textarea",
                "name": "content",
                "onBlur": [Function],
                "onChange": [Function],
                "placeholder": "textarea",
                "rows": "7",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="form-check flex-container__b"
                >
                  <div
                    className="private__comment"
                  >
                    Make it Private
                  </div>
                  <div
                    className="comment_check"
                  >
                    <input
                      className="form-check"
                      name="checkbox"
                      onChange={[Function]}
                      type="checkbox"
                      value={true}
                    />
                  </div>
                </div>,
                <Button
                  Class=""
                  className="comment-button"
                  label="Add Comment"
                  onClick={[Function]}
                  type="comment"
                />,
              ],
              "className": "flex-container__a",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="private__comment"
                    >
                      Make it Private
                    </div>,
                    <div
                      className="comment_check"
                    >
                      <input
                        className="form-check"
                        name="checkbox"
                        onChange={[Function]}
                        type="checkbox"
                        value={true}
                      />
                    </div>,
                  ],
                  "className": "form-check flex-container__b",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Make it Private",
                      "className": "private__comment",
                    },
                    "ref": null,
                    "rendered": "Make it Private",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <input
                        className="form-check"
                        name="checkbox"
                        onChange={[Function]}
                        type="checkbox"
                        value={true}
                      />,
                      "className": "comment_check",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-check",
                        "name": "checkbox",
                        "onChange": [Function],
                        "type": "checkbox",
                        "value": true,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "Class": "",
                  "className": "comment-button",
                  "label": "Add Comment",
                  "onClick": [Function],
                  "type": "comment",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="comment__head"
          >
            <p
              className="comment__head__a"
            >
              ADD COMMENT
            </p>
            <p
              className="comment__head__b"
            >
              Comment
            </p>
          </div>,
          <div>
             
            <div />
          </div>,
          <form
            className=""
            onSubmit={[Function]}
          >
            <FormGroup
              bsClass="form-group"
              controlId="formControlsTextarea"
            >
              <FormControl
                bsClass="form-control"
                className="comment-form"
                cols="70"
                componentClass="textarea"
                name="content"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="textarea"
                rows="7"
                value=""
              />
            </FormGroup>
            <div
              className="flex-container__a"
            >
              <div
                className="form-check flex-container__b"
              >
                <div
                  className="private__comment"
                >
                  Make it Private
                </div>
                <div
                  className="comment_check"
                >
                  <input
                    className="form-check"
                    name="checkbox"
                    onChange={[Function]}
                    type="checkbox"
                    value={true}
                  />
                </div>
              </div>
              <Button
                Class=""
                className="comment-button"
                label="Add Comment"
                onClick={[Function]}
                type="comment"
              />
            </div>
          </form>,
        ],
        "className": "container form__comment",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="comment__head__a"
              >
                ADD COMMENT
              </p>,
              <p
                className="comment__head__b"
              >
                Comment
              </p>,
            ],
            "className": "comment__head",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ADD COMMENT",
                "className": "comment__head__a",
              },
              "ref": null,
              "rendered": "ADD COMMENT",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Comment",
                "className": "comment__head__b",
              },
              "ref": null,
              "rendered": "Comment",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              false,
              <div />,
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FormGroup
                bsClass="form-group"
                controlId="formControlsTextarea"
              >
                <FormControl
                  bsClass="form-control"
                  className="comment-form"
                  cols="70"
                  componentClass="textarea"
                  name="content"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="textarea"
                  rows="7"
                  value=""
                />
              </FormGroup>,
              <div
                className="flex-container__a"
              >
                <div
                  className="form-check flex-container__b"
                >
                  <div
                    className="private__comment"
                  >
                    Make it Private
                  </div>
                  <div
                    className="comment_check"
                  >
                    <input
                      className="form-check"
                      name="checkbox"
                      onChange={[Function]}
                      type="checkbox"
                      value={true}
                    />
                  </div>
                </div>
                <Button
                  Class=""
                  className="comment-button"
                  label="Add Comment"
                  onClick={[Function]}
                  type="comment"
                />
              </div>,
            ],
            "className": "",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": <FormControl
                  bsClass="form-control"
                  className="comment-form"
                  cols="70"
                  componentClass="textarea"
                  name="content"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="textarea"
                  rows="7"
                  value=""
                />,
                "controlId": "formControlsTextarea",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "comment-form",
                  "cols": "70",
                  "componentClass": "textarea",
                  "name": "content",
                  "onBlur": [Function],
                  "onChange": [Function],
                  "placeholder": "textarea",
                  "rows": "7",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="form-check flex-container__b"
                  >
                    <div
                      className="private__comment"
                    >
                      Make it Private
                    </div>
                    <div
                      className="comment_check"
                    >
                      <input
                        className="form-check"
                        name="checkbox"
                        onChange={[Function]}
                        type="checkbox"
                        value={true}
                      />
                    </div>
                  </div>,
                  <Button
                    Class=""
                    className="comment-button"
                    label="Add Comment"
                    onClick={[Function]}
                    type="comment"
                  />,
                ],
                "className": "flex-container__a",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="private__comment"
                      >
                        Make it Private
                      </div>,
                      <div
                        className="comment_check"
                      >
                        <input
                          className="form-check"
                          name="checkbox"
                          onChange={[Function]}
                          type="checkbox"
                          value={true}
                        />
                      </div>,
                    ],
                    "className": "form-check flex-container__b",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Make it Private",
                        "className": "private__comment",
                      },
                      "ref": null,
                      "rendered": "Make it Private",
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <input
                          className="form-check"
                          name="checkbox"
                          onChange={[Function]}
                          type="checkbox"
                          value={true}
                        />,
                        "className": "comment_check",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "form-check",
                          "name": "checkbox",
                          "onChange": [Function],
                          "type": "checkbox",
                          "value": true,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "Class": "",
                    "className": "comment-button",
                    "label": "Add Comment",
                    "onClick": [Function],
                    "type": "comment",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
